repos/system_upgrade/common/actors/checkfstabmountorder/libraries/checkfstabmountorder.py

```diff
@@ -0,0 +1,51 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def _get_incorrectly_ordered_fstab_entries(fstab):
+    """
+    Retrieve pairs of incorrectly ordered entries in /etc/fstab based on mount point overshadowing.
+
+    :param list[FstabEntry] fstab: fstab entries from StorageInfo
+    :returns: fstab entries with incorrect order
+    """
+
+    for i, fstab_entry in enumerate(fstab):
+        mount_point = fstab_entry.fs_file
+
+        for overshadowing_fstab_entry in fstab[i:]:
+            overshadowing_mount_point = overshadowing_fstab_entry.fs_file
+            if mount_point != overshadowing_mount_point and mount_point.startswith(overshadowing_mount_point):
+                yield fstab_entry, overshadowing_fstab_entry
+
+
+def check_fstab_mount_order():
+    critical_mount_points = ['/var/log', '/var/lib', '/var/lib/leapp']
```

Comment:
we should not care about critical mountpoints in this case. the very same problem we could find in other cases, e.g.
/usr/lib
/usr
...

and all other partitions that exists on the system. the actor is supposed to ensure, that all entries in fstab are in the correct order.

Reply:
Thanks for clearing that up to me, it's fixed now.